api/src/opentrons/protocol_api/labware.py

```
@@ -47,6 +49,38 @@ class OutOfTipsError(Exception):
     pass
 
 
+# TODO: AA 2020-06-10 move out of protocol_api
+class TipLengthCalNotFound(Exception):
+    pass
+
+
+# TODO: AA 2020-06-10 move out of protocol_api
+class DateTimeEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, datetime.datetime):
+            return obj.isoformat()
+        return json.JSONEncoder.default(self, obj)
+
+
+# TODO: AA 2020-06-10 move out of protocol_api
+class DateTimeDecoder(json.JSONDecoder):
+    def __init__(self):
+        super().__init__(object_hook=self.dict_to_obj)
+
+    def dict_to_obj(self, d):
+        if isinstance(d, dict):
+            d = {k: self._decode_datetime(v) for k, v in d.items()}
+        return d
+
+    def _decode_datetime(self, obj):
+        try:
+            return datetime.datetime.fromisoformat(obj)
+        except ValueError:
+            return obj
```

I think we should probably let these raise, otherwise we'll get some _very_ strange type confusions out of misformatted dicts. We should handle the possible errors in the decode, and just remove the calibration.

---

Oh, I didn't understand that swallowing these errors is required because they get called on all values. Let's keep swallowing the errors, but assert that the value decoded properly afterward.